src/meta/api/src/schema_api_impl.rs

```diff
@@ -1715,6 +1725,267 @@ impl<KV: KVApi> SchemaApi for KV {
         ))
     }
 
+    async fn get_table_stage_file_info(
+        &self,
+        req: GetTableStageFileReq,
+    ) -> Result<GetTableStageFileReply, MetaError> {
+        debug!(req = debug(&req), "SchemaApi: {}", func_name!());
+
+        let tenant_dbname_tbname = &req.table;
+        let tenant_dbname = tenant_dbname_tbname.db_name_ident();
+
+        // Get db by name to ensure presence
+
+        let (db_id_seq, db_id) = get_u64_value(self, &tenant_dbname).await?;
+        debug!(db_id_seq, db_id, ?tenant_dbname_tbname, "get database");
+
+        db_has_to_exist(
+            db_id_seq,
+            &tenant_dbname,
+            format!("get_table: {}", tenant_dbname_tbname),
+        )?;
+
+        // Get table by tenant,db_id, table_name to assert presence.
+
+        let dbid_tbname = DBIdTableName {
+            db_id,
+            table_name: tenant_dbname_tbname.table_name.clone(),
+        };
+
+        let (tb_id_seq, table_id) = get_u64_value(self, &dbid_tbname).await?;
+        table_has_to_exist(tb_id_seq, tenant_dbname_tbname, "get_table")?;
+
+        let tbid = TableId { table_id };
+
+        let (tb_meta_seq, tb_meta): (_, Option<TableMeta>) = get_struct_value(self, &tbid).await?;
+
+        table_has_to_exist(
+            tb_meta_seq,
+            tenant_dbname_tbname,
+            format!("get_table meta by: {}", tbid),
+        )?;
+
+        debug!(
+            ident = display(&tbid),
+            name = display(&tenant_dbname_tbname),
+            table_meta = debug(&tb_meta),
+            "get_table_stage_file_info"
+        );
+
+        let mut file_info = BTreeMap::new();
+
+        for file in req.files {
+            let key = TableStageFileNameIdent {
+                tenant: tenant_dbname_tbname.tenant.clone(),
+                db_id,
+                table_id,
+                file: file.clone(),
+            };
+
+            let (_file_seq, file_stage_info): (_, Option<TableStageFileInfo>) =
+                get_struct_value(self, &key).await?;
+
+            if let Some(file_stage_info) = file_stage_info {
+                file_info.insert(file, file_stage_info);
+            }
+        }
+
+        Ok(GetTableStageFileReply { file_info })
```

Comment:
This is a read operation without a consistency guarantee:

The table may be moved to another db with `rename_table`.
The files may be modified while reading every one of them.

I can not tell if this logic will work correctly, it depends on whether its caller can tolerate these inconsistencies.

Reply:
yes, i think `copy into` can tolerate these inconsistencies.